source/common/http/http1/codec_impl.cc

```diff
@@ -473,10 +475,22 @@ void ConnectionImpl::dispatch(Buffer::Instance& data) {
   ssize_t total_parsed = 0;
   if (data.length() > 0) {
     for (const Buffer::RawSlice& slice : data.getRawSlices()) {
-      total_parsed += dispatchSlice(static_cast<const char*>(slice.mem_), slice.len_);
+      dispatching_ = true;
+      auto result = dispatchSlice(static_cast<const char*>(slice.mem_), slice.len_);
```

Comment:
It would be nice to use StatusOr<size_t> as a return value from dispatchSlice and have it return codec_exception_ otherwise.

Reply:
+1. However we will need our own version of StatusOr<> if we have custom Status class with our own error space that maps to existing exception types.

Reply:
Done